source/Public/SQL Database/Get-FabricSQLDatabase.ps1

```diff
@@ -74,20 +78,24 @@ function Get-FabricSQLDatabase {
     )
 
     begin {
+
         Confirm-TokenState
 
-        if ($PSBoundParameters.ContainsKey("SQLDatabaseName") -and $PSBoundParameters.ContainsKey("SQLDatabaseId")) {
-            throw "Parameters SQLDatabaseName and SQLDatabaseId cannot be used together"
-        }
     }
 
     process {
+
+        if ($PSBoundParameters.ContainsKey("SQLDatabaseName") -and $PSBoundParameters.ContainsKey("SQLDatabaseId")) {
```

Comment:
Shouldn't this validation stay in "begin" block, as long as the validation needs to be run only once rather than multiple times for each instance of the function? Just checking, because that was my understanding of begin/process/end.
[about_Functions_Advanced_Methods](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_functions_advanced_methods?view=powershell-7.5)

Reply:
It depends on the actions that you need to take with the validation. 

It is now in the same place as the parameter set validation, so I would think you either have both in the begin or both in the process.

Reply:
I don't follow your logic.  There will not be a mixture of Workspace Name and Workspace id passed into the function so that validation will be done every time in the same object type.  As I said,  either piece of logic can go in either block.  They were I'm separate blocks,  which was fine. I moved them into the same block, which is also fine.

Reply:
Agree, there will not be a mixture of `WorkspaceName` and `WorkspaceId`, because the first doesn't exist.
If you meant mixture of `SQLDatabaseName` and `SQLDatabaseId` - this can happen; hence validation in `begin` block  (these parameters never come from pipeline) and should be check only once.
ParameterSetName checks `WorkspaceObject` (pipe) and must be executed for each iteration to determine `$WorkspaceId` when pipe is provided (or do nothing when `$WorkspaceId` is provided and no pipe).
Maybe you are confused `WorkspaceId` with `SQLDatabaseId` and hence this misalignment?
Finally, we don't want to see X warnings about the same issue? (screenshot below)
![image](https://github.com/user-attachments/assets/d6ecd689-8f76-4a0c-baff-f7c593dd23d7)

Reply:
Oh thats very interesting. 

Could you share the code you are using that gives that error?

because I cannot reroduce